src/components/ScrollLock/tests/ScrollLock.test.tsx

```diff
@@ -29,23 +29,29 @@ describe('ScrollLock', () => {
 
     scrollLockContainer.instance().setState({showScrollLock: false});
 
-    expect(document.body.hasAttribute(`${SCROLL_LOCKING_ATTRIBUTE}`)).toBe(
-      true,
-    );
+    expect(
+      document.body.firstElementChild.hasAttribute(
+        `${SCROLL_LOCKING_WRAPPER_ATTRIBUTE}`,
+      ),
+    ).toBe(true);
   });
 
-  it('adds data attribute to the body when it mounts', () => {
+  it('adds data attribute to the first child element of the body when it mounts', () => {
     mountWithAppProvider(<ScrollLock />);
-    expect(document.body.hasAttribute(`${SCROLL_LOCKING_ATTRIBUTE}`)).toBe(
-      true,
-    );
+    expect(
+      document.body.firstElementChild.hasAttribute(
+        `${SCROLL_LOCKING_WRAPPER_ATTRIBUTE}`,
+      ),
+    ).toBe(true);
   });
 
-  it('removes data attribute from the body when it unmounts', () => {
+  it('removes data attribute from the first child element of the body when it unmounts', () => {
     const scrollLock = mountWithAppProvider(<ScrollLock />);
     scrollLock.unmount();
-    expect(document.body.hasAttribute(`${SCROLL_LOCKING_ATTRIBUTE}`)).toBe(
-      false,
-    );
+    expect(
+      document.body.firstElementChild.hasAttribute(
```

Comment:
I’d like to write a test for this, but it fails. I’m not familiar enough with Jest/Enzyme/React to understand why this is failing (this might be why it wasn’t tested at all). Any pointers?

Reply:
This worked for me. There was no first child so I created one.
```
describe('ScrollLock', () => {
  it('does not remove the data attribute from the first child element of the body when two scrolllocks are mounted and one unmounts', () => {
    class DummyFrame extends React.Component {
      state = {
        showScrollLock: true,
      };

      render() {
        const {showScrollLock} = this.state;

        // eslint-disable-next-line shopify/jest/no-if
        const scrollLockMarkup = showScrollLock ? <ScrollLock /> : null;

        return (
          <React.Fragment>
            {scrollLockMarkup}
            <ScrollLock />
          </React.Fragment>
        );
      }
    }

    document.body.appendChild(document.createElement('div'));
    const scrollLockContainer = mountWithAppProvider(<DummyFrame />);

    scrollLockContainer.instance().setState({showScrollLock: false});

    const lockedWrapper = document.querySelector(
      `[${SCROLL_LOCKING_WRAPPER_ATTRIBUTE}]`,
    );

    expect(lockedWrapper).toBeTruthy();
  });

  it('adds data attribute to the first child element of the body when it mounts', () => {
    document.body.appendChild(document.createElement('div'));
    mountWithAppProvider(<ScrollLock />);
    const lockedWrapper = document.querySelector(
      `[${SCROLL_LOCKING_WRAPPER_ATTRIBUTE}]`,
    );
    expect(lockedWrapper).toBeTruthy();
  });

  it('removes data attribute from the first child element of the body when it unmounts', () => {
    document.body.appendChild(document.createElement('div'));
    const scrollLock = mountWithAppProvider(<ScrollLock />);
    scrollLock.unmount();
    const lockedWrapper = document.querySelector(
      `[${SCROLL_LOCKING_WRAPPER_ATTRIBUTE}]`,
    );
    expect(lockedWrapper).toBeFalsy();
  });
});
```